misaligned.cpp

```diff
@@ -1,21 +1,70 @@
 #include <assert.h>
 #include <iostream>
+#include <sstream>
+#include <string>
+#include <algorithm>
+#include <tuple>
 
-int printColorMap() {
+int printColorMap(std::ostream& out) {
     const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
     const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
     for (i = 0; i < 5; i++) {
         for (j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            out << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
         }
     }
     return i * j;
 }
 
+// Test - core part
+std::tuple<int, int, int, int> testPrintColorMap() {

```

Comment:
The signature of the function does not say the order in which things are being returned. Better to use a struct with named members, to avoid misunderstandings between this function and its callers